interfaces/cython/cantera/ck2yaml.py

```diff
@@ -2093,16 +2064,36 @@ def convert_mech(input_file, thermo_file=None, transport_file=None,
         surface_names = parser.write_yaml(name=phase_name, out_name=out_name)
         if not quiet:
             nReactions = len(parser.reactions) + sum(len(surf.reactions) for surf in parser.surfaces)
-            print('Wrote YAML mechanism file to {0!r}.'.format(out_name))
-            print('Mechanism contains {0} species and {1} reactions.'.format(len(parser.species_list), nReactions))
-        return surface_names
+            logger.info('Wrote YAML mechanism file to {0!r}.'.format(out_name))
+            logger.info('Mechanism contains {0} species and {1} reactions.'.format(
+                        len(parser.species_list), nReactions))
+        return parser, surface_names
+
+    def show_duplicate_reactions(self, error_message):
+        reactions = []
+        for line in error_message.split('\n'):
+            match = re.match('>.*# Reaction ([0-9]+)', line)
+            if match:
+                reactions.append(int(match.group(1))-1)
+
+        if len(reactions) != 2:
```

Comment:
Why is this check necessary? From my quick test, adding a third duplicate reaction still only shows two of them. Are there other error conditions that can reach this function?

Reply:
Right, the error from `Kinetics::checkDuplicates` should only ever indicate two reactions. What I'm trying to hedge against is (a) the possibility there's some case where that error message doesn't have the expected format, or (b) some future change breaks this attempt at parsing the error message. If either of those happens, I'd rather output the original message from `Kinetics::checkDuplicates` than some error message generated caused by `reactions` not having the the expected number of entries.

Reply:
I added some comments to `show_duplicate_reactions` which will hopefully make some of this clear to our future selves. 😂